web/src/lib.rs

```diff
@@ -771,6 +776,21 @@ impl RuffleHandle {
                 },
             ));
 
+            // Create webglcontextlost handler.
+            instance.webglcontextlost_callback = Some(JsCallback::register(
+                &player.canvas,
+                "webglcontextlost",
+                false,
+                move |_js_event: WebGlContextEvent| {
+                    INSTANCES.with(|instances| {
+                        if instances.borrow().len() >= 8 {
+                            let _ = ruffle.remove_instance();
+                            js_player.reload_with_canvas_renderer();
```

Comment:
to be sure, how come we didn't need to `js_player.clone()` anywhere?
Does this mean a clone wasn't needed somewhere earlier already?

Reply:
Not entirely sure as I'm not super familiar with Rust, but isn't it because:

1) I use the move keyword on the closures scope
2) The `reload_with_canvas_renderer` wasm-bindgen function uses it by reference.

Reply:
That's a good question though, I can switch to the method used here:

https://github.com/ruffle-rs/ruffle/blob/ba677e0c40a74488be911eb8f2ba2445a1641f7b/web/src/lib.rs#L593-L599

Just not sure if I have to do so.

Reply:
The `js_player.clone()` in the `mouse_down_callback` does seem to be needed, perhaps because of the `let _ = ruffle.with_instance(move |instance| {`